src/ophyd_async/core/signal.py

```diff
@@ -557,6 +562,63 @@ async def wait_for_value(
     await checker.wait_for_value(signal, timeout)
 
 
+async def set_and_wait_for_other_value(
+    set_signal: SignalRW[T],
+    set_value: T,
+    read_signal: SignalR[S],
+    read_value: S,
+    timeout: float = DEFAULT_TIMEOUT,
+    set_timeout: Optional[float] = None,
+) -> AsyncStatus:
+    """Set a signal and monitor another signal until it has the specified value.
+
+    This function sets a set_signal to a specified set_value and waits for
+    a read_signal to have the read_value.
+
+    Parameters
+    ----------
+    signal:
+        The signal to set
+    set_value:
+        The value to set it to
+    read_signal:
+        The signal to monitor
+    read_value:
+        The value to wait for
+    timeout:
+        How long to wait for the signal to have the value
+    set_timeout:
+        How long to wait for the set to complete
+
+    Notes
+    -----
+    Example usage::
+
+        set_and_wait_for_value(device.acquire, 1, device.acquire_rbv, 1)
+    """
+    # Start monitoring before the set to avoid a race condition
+    values_gen = observe_value(read_signal)
+
+    # Get the initial value from the monitor to make sure we've created it
+    await anext(values_gen)
```

Comment:
One last thing to consider. If we have:
- `signal.value == 3`
- `await set_and_wait_for_value(signal, 3)`
- MDEL>=0

Then this code will receive the initial value, discard it, set it to the value it currently is, then timeout waiting for an update that never comes.

Possible fixes:
1. Don't bother setting the signal if it's already at the right value
2. Error if the signal is already at the right value
3. Store `current_value = await anext(values_gen)`, then only call `_wait_for_value` if it doesn't match

I think I favour 3, what do you think?

Reply:
I agree that 3 seems the most broadly correct, small enough that we can do as part of this I think. Will do now